tests/test_connection.py

```diff
@@ -341,3 +349,274 @@ def cb1(n):
 
     with pytest.raises(ValueError):
         conn.remove_notify_handler(cb1)
+
+
+def test_transaction(conn):
+    """Basic use of transaction() to BEGIN and COMMIT a transaction."""
+    assert conn.pgconn.transaction_status == conn.TransactionStatus.IDLE
+    with conn.transaction():
+        assert conn.pgconn.transaction_status == conn.TransactionStatus.INTRANS
+    assert conn.pgconn.transaction_status == conn.TransactionStatus.IDLE
+
+
+def test_transaction_exposes_associated_connection(conn):
+    """Transaction exposes it's connection as a read-only property."""
+    tx = conn.transaction()
+    assert tx.connection is conn
+    with pytest.raises(AttributeError):
+        tx.connection = conn
+
+
+def test_transaction_is_not_kept_alive_by_connection(conn):
+    """
+    Since the Transaction has a reference back to the connection, ensure that
+    the opposite is not also true, as we want to avoid reference cycles.
```

Comment:
I don't feel we need this test, if we are not creating a reference cycle. The Connection only creates a Transaction, it doesn't hold a pointer to it, right?

Reply:
Yeah this is basically a test to fail if a reference cycle is ever introduced in the future. Happy to remove if you think it has little value.

Reply:
No, I don't think it has much value. There could be potentially cycles between any pair of objects, between connections and cursors for instance, but unless you really start referring to each other you don't really know the scenarios in which a cycle may be created. I think it's better addressed when you know when a cycle could exist.